### PR TITLE
fix one big and 2 small bugs in training

### PR DESCRIPTION
**Pull request by bkal01**
steps were being computed incorrectly so the LR scheduler was configured wrong. this means that the entire llava-pretrain run would just be in warmup

fixed some logging/saving logic to be more aligned with actual optimizer steps

added seed for reproducibility

fixed LR scheduler configuration on resuming training

### CODE REVIEW

**Review by sanggusti — APPROVED** [review]
Nice catch. Let's merge this
